packages/router/src/params.tsx

```diff
@@ -16,29 +14,27 @@ export const ParamsProvider: React.FC = ({ children }) => {
   const { routes, paramTypes } = useRouterState()
   const location = useLocation()
 
-  let initialParams = parseSearch(location.search)
+  let params = parseSearch(location.search)
 
   for (const route of routes) {
     if (route.path) {
-      const { match, params } = matchPath(
+      const { match, params: pathParams } = matchPath(
         route.path,
         location.pathname,
         paramTypes
       )
 
       if (match) {
-        initialParams = {
-          ...initialParams,
+        params = {
           ...params,
+          ...pathParams,
         }
       }
     }
   }
 
-  const [params, setParams] = useState<Record<string, string>>(initialParams)
-
   return (
-    <ParamsContext.Provider value={{ params, setParams }}>
+    <ParamsContext.Provider value={{ params }}>
```

Comment:
This will now update the params based off of state / location updates instead of setting the state in nested routes